compiler/src/iree/compiler/ConstEval/Runtime.cpp

```diff
@@ -94,11 +152,126 @@ void CompiledBinary::deinitialize() {
   device.reset();
 }
 
-LogicalResult CompiledBinary::invokeNullary(Location loc, StringRef name,
-                                            ResultsCallback callback) {
+FunctionCall::FunctionCall(CompiledBinary &binary, iree_host_size_t argCapacity,
+                           iree_host_size_t resultCapacity)
+    : binary(binary) {
+  IREE_CHECK_OK(iree_vm_list_create(iree_vm_make_undefined_type_def(),
+                                    argCapacity, iree_allocator_system(),
+                                    &inputs));
+  IREE_CHECK_OK(iree_vm_list_create(iree_vm_make_undefined_type_def(),
+                                    resultCapacity, iree_allocator_system(),
+                                    &outputs));
+}
+
+// Imports or snapshots a raw host buffer, depending on whether import is
+// possible.
+LogicalResult FunctionCall::importBufferForRead(Location loc,
+                                                const uint8_t *rawData,
+                                                iree_host_size_t length,
+                                                iree_hal_buffer_t **buffer) {
+  // TODO: Allow import when we have resources in the input where alignment
+  // can be guaranteed.
+  bool tryImport = false;
+  if (tryImport) {
+    iree_hal_buffer_params_t params;
+    std::memset(&params, 0, sizeof(params));
+    params.type = IREE_HAL_MEMORY_TYPE_OPTIMAL_FOR_DEVICE;
+    iree_hal_external_buffer_t external_buffer;
+    std::memset(&external_buffer, 0, sizeof(external_buffer));
+    external_buffer.type = IREE_HAL_EXTERNAL_BUFFER_TYPE_HOST_ALLOCATION;
+    external_buffer.size = length;
+    external_buffer.handle.host_allocation.ptr =
+        const_cast<void *>(static_cast<const void *>(rawData));
+    auto status = iree_hal_allocator_import_buffer(
+        binary.getAllocator(), params, &external_buffer,
+        /*release_callback=*/{nullptr, nullptr}, buffer);
+    if (iree_status_is_ok(status))
+      return success();
+    else if (!(iree_status_is_out_of_range(status) ||
+               iree_status_is_unavailable(status)))
+      return failure();
```

Comment:
leaking the status here; return handleRuntimeError I think?

Reply:
Oops. Fixed.